bolides/bdf.py

```diff
@@ -0,0 +1,1032 @@
+
+        elif source == 'remote':
+            if 'url' not in kwargs:
+                raise ValueError('Must specify a url with url=...')
+            url = kwargs['url']
+            init_gdf = remote(url)
+
+        elif source == 'usg-orbits':
+            init_gdf = remote('https://aozerov.com/data/usg-orbits.csv')
+            annotate = False
+
+        # elif source == 'glm-orbits':
+        #     init_gdf = csv(file=ROOT_PATH+'/../notebooks/glm-orbits.csv')
+        #     annotate = False
+
+        elif source in ['glm-pipeline', 'pipeline']:
+            source = 'glm-pipeline'
+            if 'min_confidence' in kwargs:
+                min_confidence = kwargs['min_confidence']
+            else:
+                min_confidence = 0
+            init_gdf = pipeline(files=files, min_confidence=min_confidence)
+
+        init_gdf['source'] = source
+
+        # rearrange columns, respecting original order if csv or pickle
+        if source not in ['csv', 'pickle'] or rearrange is True:
+            first_cols = [col for col in _FIRST_COLS if col in init_gdf.columns]
+            other_cols = [col for col in init_gdf.columns if col not in first_cols]
+            init_gdf = init_gdf[first_cols + other_cols]
+
+        # initialize the super-class (GeoDataFrame) using the created init_gdf
+        super().__init__(init_gdf)
+
+        # add additional metadata to the bolides
+        if annotate:
+            self.annotate()
+
+        self.descriptions = pd.read_csv(ROOT_PATH+'/metadata/columns.csv', index_col='column')
+
+    def annotate(self):
+        """Add metadata to bolide detections"""
+        from .astro_utils import get_phase, get_solarhour, get_sun_alt
+
+        # lunar phase
+        self['phase'] = [get_phase(dt) for dt in self['datetime']]
+        # moon fullness
+        self['moon_fullness'] = -np.abs(self['phase']-0.5)*2+1
+        # solar hour
+        self['solarhour'] = [get_solarhour(data[0], data[1]) for data in zip(self['datetime'], self['longitude'])]
+        # solar altitude
+        sun_alt = np.array([get_sun_alt(dt=row['datetime'].to_pydatetime(), lat=row['latitude'], lon=row['longitude']) for _, row in self.iterrows()])
+        self['sun_alt_obs'] = sun_alt[:, 0]
+        self['sun_alt_app'] = sun_alt[:, 1]
+
+        self['date_retrieved'] = datetime.now()
+
+    def describe(self, key=None):
+        """Describe a variable of the BolideDataFrame
+
+        Parameters
+        ----------
+        key : str, iterable, or None
+            The variables to be described.
+            If str, describes that variable.
+            If iteralbe, describe all variable strings in the iterable.
+            If None, describe all variables in the BolideDataFrame.
+
+        Returns
+        -------
+        str :
+            A string describing the variable(s).
+        """
+
+        if type(key) is str:
+            key = [key]
+        to_describe = self.columns if key is None else key
```

Comment:
In the tutorial you say either use `bdf.columns` or bdf['columns']. The latter throws and error.

Reply:
Clarified this by using a hypothetical column named `interesting_data` in the tutorial.